command/s3_pull.go

```diff
@@ -25,7 +24,7 @@ type s3Base struct {
 	ExcludeFilter string `mapstructure:"exclude" plugin:"expand"`
 	MaxRetries    uint   `mapstructure:"max_retries"`
 
-	bucket pail.Bucket
+	bucket pail.SyncBucket
```

Comment:
why are these different?

Reply:
Mostly because the archive bucket hasn't implemented anything sane for single-file operations (i.e. Get/Put/Upload/Download), since those would require you to stream the entire archive file just to add/get a single file. The SyncBucket has just the Push/Pull functionality since the archive bucket is only good at whole-directory operations and bad at most other things.

Reply:
fair. I think it's ok if they all share the same interface, and leave it to users to choose the right implementation for their use case (particularly if the implementations have good fall back behavior), but this is fine for now.